contrib/dracut/systemd/zfs-dracut-bootfs.service.in

```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Set BOOTFS environment for dracut
+Documentation=man:zfs(8)
+DefaultDependencies=no
+After=zfs-import.target
+Requires=zfs-import.target
+
+[Service]
+Type=oneshot
+ExecStart=/bin/sh -c "/bin/systemctl set-environment BOOTFS=$(@sbindir@/zpool list -H -o bootfs | grep -m1 -v '^-$')"
```

Comment:
I imagine the awk as used before may be faster / better not too sure.  It was a while ago, but I do remember trying things like grep.  It's possible I just didn't find the right combo.

Reply:
~~~
c0d3@z3r0:[~]# time echo -e '-\n-\nrpool/ROOT\ndpool/ROOT' | grep -m1 -v '^-$'
rpool/ROOT

real	0m0,003s
user	0m0,002s
sys	0m0,001s
c0d3@z3r0:[~]# time echo -e '-\n-\nrpool/ROOT\ndpool/ROOT' | awk '$1 != "-" {print; exit}'
rpool/ROOT

real	0m0,004s
user	0m0,003s
sys	0m0,001s
~~~

There should be no noticable difference in boot speed. That is absolutely negligible :-)